### DISCUSSION TITLE
Cannot build bundle js when specified publicPath in dev mode for webpack 5

### DISCUSSION THREAD

**Discussion by mebest100** [webpack/webpack#14656] [Q&A]
# Bug report

**What is the current behavior?**

when I specified publicPath at "/test/", but found html no display when run webpack-dev-server,and furthermore found no bundle js produced ! (observed it from Chrome source Tab) . 
I can only resovle it that I must manually specified the bundle js by scripts label in index.html template just like that:
` <script type="text/javascript" src="/test/main.js"></script>`

This is very strangely and not reasonable.

**If the current behavior is a bug, please provide the steps to reproduce.**
Yes, I'm sure this is bug , and I tried many time in different cases.

**What is the expected behavior?**
Pls confirm from your side and try to fix it.

**Other relevant information:**
webpack version:  5.61.0
Node.js version: 12.18.1
Operating System: 
Additional tools:

**Comment by alexander-akait** [comment]
Please create reproducible test repo

**Reply by mebest100** [reply]
Pls check below reproducible test repo:
https://github.com/mebest100/webpack-bugreproduce.git

**Comment by alexander-akait** [comment]
Expected, you set `/test/` public path, but do emit file in `test` directory, why do you set it on `/test/`, default value is `auto` it is good default choice. You can set:
```
new HtmlWebpackPlugin({
  template: path.resolve(__dirname, "./public/index.html"),
  filename: "index.html",
  publicPath: ''
})
```

Please read https://webpack.js.org/configuration/output/#outputpublicpath

**Reply by mebest100** [reply]
But this problem only ocurred in dev mode, and there is no such problem in Prod Env and it can produce bundle js even when I specified pulicPath at /test/ in prod Env.
Furthermoe the reason I set pulicPath at /test/ is I want to find out what publicPath is for and what it can do and it's real meaning since you know  webpack's official doc for publicPath is really very bad and poor clear.
According to my understanding and test result, I conclude that publicPath is just a virtual path which map to the physical storage location root of served html files on webserver, it very similar to express's static mapping path.
And actually I found that I must config location alias in nginx for publicPath which also map to html root diretory when deploy to prod Env otherwise browser cannot find css or other files prefixed by publicPath.

![image](https://user-images.githubusercontent.com/11530991/140627274-fe6c1407-a505-46e3-a274-78032cdf2817.png)


But based your comments, html-webpack-plugin will emit files to publicPath,no need to manually config it and keep default is ok, its seems publicPath is no useful and not meaningful。
I totally get confused what is  publicPath for and its existence is meaningful?

**Reply by alexander-akait** [reply]
Please use stackoverflow for these questions, public path is just url, it is not affected on file system structure
